c-pallets/segment-book/src/lib.rs

```diff
@@ -843,6 +857,7 @@ pub mod pallet {
 				if value.len() as u32 > max_len {
 					max_len = value.len() as u32;
 				}
+				// why offchain_sign_digest inside the for loop?
```

Comment:
Because I used the same signature in the process of trying, calling multiple trade fairs failed, and one signature must correspond to one transaction, so I made several `offchains_sign_digest()` operation.

Reply:
I need to store a transaction for each miner's challenge, store each miner's challenge in a loop, and I need a new signature for each transaction